R/container.R

```diff
@@ -0,0 +1,174 @@
+#' Declare post-processing for model predictions
+#'
+#' @param mode The model's mode, one of `"unknown"`, `"classification"`, or
+#' `"regression"`. Modes of `"censored regression"` are not currently supported.
+#' @param type The model sub-type. Possible values are `"unknown"`, `"regression"`,
+#' `"binary"`, or `"multiclass"`.
+#' @param outcome The name of the outcome variable.
+#' @param estimate The name of the point estimate (e.g. predicted class)
+#' @param probabilities The names of class probability estimates (if any). For
+#' classification, these should be given in the order of the factor levels of
+#' the `estimate`.
+#' @param time The name of the predicted event time. (not yet supported)
+#' @param call The call to be displayed in warnings or errors.
+#' @examples
+#'
+#' container()
+#' @export
+container <- function(mode = "unknown", type = "unknown", outcome = character(0),
```

Comment:
We’ll probably need to validate the type of model as well as it “species.” Specifically, we might need to differentiate between binary and multiclass classification. 

We can do this via an argument/attribute or by adding a more specific class to the container.

Reply:
I'd prefer that users need not specify either of `mode` or `type` when using containers with a workflow. It probably makes sense for `container::fit.container()` to continue requiring these be set before running `fit.container()`; `workflows::fit.action_container()` could set `mode` and `type` at `workflows::fit.workflow()` time before dispatching off to `fit.container()`.

Reply:
Yes, this is what I thought that we would do.